test/Infrastructure.IntegrationTest/Repositories/EmergencyAccessRepositoryTests.cs

```diff
@@ -15,15 +15,15 @@ public class EmergencyAccessRepositoriesTests
         var grantorUser = await userRepository.CreateAsync(new User
         {
             Name = "Test Grantor User",
-            Email = "test+grantor@email.com",
+            Email = $"test+grantor{Guid.NewGuid()}@email.com",
```

Comment:
This guid makes me suspicious -- are these tests not resetting the db to a known state?

These tests should either be run in transactions to avoid residuals, or some other means of guaranteeing isolation of tests. We can talk about specifially what/how we want to isolate, but my ideal would be on every tests. It's not great to have tests depend on execution order and clearing DBs would be a good way to enforce that.

Reply:
As we discussed a little, you are right to be suspicious, these are not clearing the database but I did work on each test happening inside a transaction but had some issues. 

As far as I can think of it's not possible to automatically do a transaction per test without making every test construct it. We could make it pretty automatic in each test class which isn't so bad. Both would be a bit of work that I can work on as a follow up. I already have 2 sub branches off this with new tests & fixes that I'd love to be able to use this PR with.

Reply:
👍 We definitely want to get there.

I think the best way to do it is to make a base class for the tests, `InfratructureTest` that stubs the methods used by the `Repository` base classes to get sql connections (dapper) or db contexts (EF) and sets up a transaction in the constructor. Then it could tear that down in dispose.

It would be a pretty minor change across all the Dapper repository files, but IIRC EF is already set up to stud db contexts.

I'm fine doing it later, though. TY